crates/db_views/src/post_view.rs

```diff
@@ -67,89 +63,145 @@ fn queries<'a>() -> Queries<
   impl ReadFn<'a, PostView, (PostId, Option<PersonId>, bool)>,
   impl ListFn<'a, PostView, PostQuery<'a>>,
 > {
-  let all_joins = |query: post_aggregates::BoxedQuery<'a, Pg>, my_person_id: Option<PersonId>| {
-    // The left join below will return None in this case
-    let person_id_join = my_person_id.unwrap_or(PersonId(-1));
+  let is_creator_banned_from_community = exists(
+    community_person_ban::table.filter(
+      post_aggregates::community_id
+        .eq(community_person_ban::community_id)
+        .and(community_person_ban::person_id.eq(post_aggregates::creator_id)),
+    ),
+  );
+
+  let is_saved = |person_id| {
+    exists(
+      post_saved::table.filter(
+        post_aggregates::post_id
+          .eq(post_saved::post_id)
+          .and(post_saved::person_id.eq(person_id)),
+      ),
+    )
+  };
+
+  let is_read = |person_id| {
+    exists(
+      post_read::table.filter(
+        post_aggregates::post_id
+          .eq(post_read::post_id)
+          .and(post_read::person_id.eq(person_id)),
+      ),
+    )
+  };
+
+  let is_creator_blocked = |person_id| {
+    exists(
+      person_block::table.filter(
+        post_aggregates::creator_id
+          .eq(person_block::target_id)
+          .and(person_block::person_id.eq(person_id)),
+      ),
+    )
+  };
+
+  let score = |person_id| {
+    post_like::table
+      .filter(
+        post_aggregates::post_id
+          .eq(post_like::post_id)
+          .and(post_like::person_id.eq(person_id)),
+      )
+      .select(post_like::score.nullable())
+      .single_value()
+  };
+
+  let all_joins = move |query: post_aggregates::BoxedQuery<'a, Pg>,
+                        my_person_id: Option<PersonId>,
+                        saved_only: bool| {
+    let is_saved_selection: Box<dyn BoxableExpression<_, Pg, SqlType = sql_types::Bool>> =
+      if saved_only {
+        Box::new(true.into_sql::<sql_types::Bool>())
+      } else if let Some(person_id) = my_person_id {
+        Box::new(is_saved(person_id))
+      } else {
+        Box::new(false.into_sql::<sql_types::Bool>())
```

Comment:
You can probably write these as `sql_types::Bool(false)` or `false.into()`.

Reply:
false